packages/kbn-content-management/table_list/src/table_list_view/services.tsx

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { FC, useContext } from 'react';
+import type { EuiTableFieldDataColumnType, SearchFilterConfig } from '@elastic/eui';
+
+import { UserContentCommonSchema } from './table_list_view';
+
+type NotifyFn = (notifyArgs: { title: string | JSX.Element; text: string }) => void;
+
+interface SavedObjectsReference {
+  type: string;
+  id: string;
+}
+
+export type DateFormatter = (props: {
+  value: number;
+  children: (formattedDate: string) => React.ReactElement;
+}) => React.ReactElement;
+
+/**
+ * Abstract external services for this component.
+ */
+export interface Services {
+  canEditAdvancedSettings: boolean;
+  getUrlForListingLimitSettings: () => string;
+  notifyError: NotifyFn;
+  getTagsColumnDefinition?: () => EuiTableFieldDataColumnType<UserContentCommonSchema> | undefined;
+  searchQueryParser?: (searchQuery: string) => {
+    searchQuery: string;
+    references?: SavedObjectsReference[];
+  };
+  getSearchBarFilters?: () => SearchFilterConfig[];
+  DateFormatterComp?: DateFormatter;
+}
+
+const TableListViewContext = React.createContext<Services | null>(null);
+
+/**
+ * Abstract external service Provider.
+ */
+export const TableListViewProvider: FC<Services> = ({ children, ...services }) => {
```

Comment:
Is there a Kibana services provider, that adapts existing contracts to this API?

e.g: https://github.com/elastic/kibana/blob/main/packages/shared-ux/card/no_data/impl/src/services.tsx#L39-L53

Reply:
Not yet. The KibanaProvider will be added when I integrate with Dashboard in a future PR... so I can see what I am doing 😊